extensions/mssql/src/objectManagement/commands.ts

```diff
@@ -111,11 +111,17 @@ async function handleObjectPropertiesDialogCommand(context: azdata.ObjectExplore
 		return;
 	}
 	try {
+		const connectedToLocalhost = context.connectionProfile.serverName === 'localhost';
 		const parentUrn = context.nodeInfo ? await getParentUrn(context) : undefined;
-		const objectType = context.nodeInfo ? context.nodeInfo.nodeType as ObjectManagement.NodeType : (context.connectionProfile.databaseName === '' ? ObjectManagement.NodeType.Server : ObjectManagement.NodeType.Database);
-		const objectName = context.nodeInfo ? context.nodeInfo.label : objectManagementLoc.PropertiesHeader;
-		const objectUrn = context.nodeInfo ? context.nodeInfo!.metadata!.urn : undefined;
-
+		let objectType = context.nodeInfo ? context.nodeInfo.nodeType as ObjectManagement.NodeType : ObjectManagement.NodeType.Server;
+		let objectName = context.nodeInfo ? context.nodeInfo.label : context.connectionProfile.serverName;
+		let objectUrn = context.nodeInfo ? context.nodeInfo!.metadata!.urn : undefined;
+		// TODO: fix connectionProfile not returning the actual server name when is localhost
+		if (!connectedToLocalhost && context.nodeInfo === undefined) {
+			objectType = context.connectionProfile.databaseName === '' ? ObjectManagement.NodeType.Server : ObjectManagement.NodeType.Database;
+			objectName = context.connectionProfile.databaseName === '' ? context.connectionProfile.serverName : context.connectionProfile.databaseName;
+			objectUrn = context.connectionProfile.databaseName === '' ? `Server[@Name='${context.connectionProfile.serverName}']` : `Server[@Name='${context.connectionProfile.serverName}']/Database[@Name='${context.connectionProfile.databaseName}']`;
```

Comment:
Make sure you're escaping `'`'s in these URNs whenever you have a property

Reply:
Can you elaborate on this?

Reply:
If you have a database with a ' in it this breaks - e.g. `My'Database`. You end up with this URN

`Database[@Name='My'Database']`

Which isn't valid since it sees the middle ' as the "end" of the string

You need to escape quote characters whenever you quote anything - this is a similar to SQL Injection.